WeakAuras/Prototypes.lua

```diff
@@ -45,6 +45,54 @@ function WeakAuras.CheckRange(unit, range, operator)
   end
 end
 
+local RangeCacheRawData = {friend = {}, harm = {}, misc = {}}
+local RangeCacheStrings = {friend = "", harm = "", misc = ""}
+local RangeTableSort = function(a, b) return a < b end
+local function RangeCacheUpdate()
+  local friend, harm, misc = RangeCacheRawData.friend, RangeCacheRawData.harm, RangeCacheRawData.misc
+  local lastvalue, friendString, harmString, miscString
+
+  twipe(friend)
+  for range in LibRangeCheck:GetFriendCheckers() do
+    tinsert(friend, range)
+  end
+  tsort(friend, RangeTableSort)
+  twipe(harm)
+  for range in LibRangeCheck:GetHarmCheckers() do
+    tinsert(harm, range)
+  end
+  tsort(harm, RangeTableSort)
+  twipe(misc)
+  for range in LibRangeCheck:GetMiscCheckers() do
+    tinsert(misc, range)
+  end
+  tsort(misc, RangeTableSort)
+
+  for _, key in pairs(friend) do
+    if lastvalue then
+      friendString = (friendString and (friendString .. ", ") or "") .. (lastvalue .. "-" .. key)
+    end
+    lastvalue = key
+  end
+  lastvalue = nil
+  for _, key in pairs(harm) do
+    if lastvalue then
+      harmString = (harmString and (harmString .. ", ") or "") .. (lastvalue .. "-" .. key)
+    end
+    lastvalue = key
+  end
+  lastvalue = nil
+  for _, key in pairs(misc) do
+    if lastvalue then
+      miscString = (miscString and (miscString .. ", ") or "") .. (lastvalue .. "-" .. key)
+    end
+    lastvalue = key
+  end
+  RangeCacheStrings.friend, RangeCacheStrings.harm, RangeCacheStrings.misc = friendString, harmString, miscString
+end
```

Comment:
There is 0 doubt in my mind that this approach will be denied, but I don't know of a better way to go about it and I don't know how else to learn but to submit my attempt.